docs/reference/service_api.yaml

```diff
@@ -0,0 +1,302 @@
+#
+# Tsuru Service API Specification
+#
+openapi: 3.0.0
+
+info:
+  title: Tsuru Service API
+  description: Tsuru Service API used to integration with external services
+  version: "1.0.0"
+
+paths:
+  /resources/plans:
+    get:
+      summary: List plans available
+      description: |
+        The service endpoint returns available service plans
+      responses:
+        200:
+          description: An array of plans
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Plan'
+        default:
+          $ref: '#/components/schemas/Error'
+  /resorces:
+    post:
+      summary: Create a new service instance
+      description: |
+        The service endpoint creates a new service instance
+      requestBody:
+        required: true
+        content:
+          application/x-www-form-urlencoded:
+            schema:
+              $ref: '#/components/schemas/InstanceForm'
+      responses:
+        201:
+          description: Instance succesfully created
+        default:
+          $ref: '#/components/schemas/Error'
+  /resources/{name}:
+    parameters:
+      - name: name
+        in: path
+        description: Instance name
+        required: true
+        schema:
+          type: string
+    get:
+      summary: Service instance additional information
+      description: |
+        The service endpoint returns additional information about the instance
+      responses:
+        404:
+          description: No additional information available
```

Comment:
Doesn't 404 here mean instance not found?

Reply:
I don't think so. Ref: https://docs.tsuru.io/stable/services/api.html#additional-info-about-an-instance
`404: when the API doesn’t have extra info about the service instance. There’s no need to include anything in the response body.`
Also: https://github.com/tsuru/tsuru/blob/master/service/endpoint.go#L371

Reply:
Ugh okay, if this is what the docs say let's keep it like this. But without this assurance we have no way to check if a given instance exists in the service. :(